var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -139,6 +147,23 @@ def cmake_args(self):
             args.append('-DADIOS2_USE_SSC={0}'.format(
                 'ON' if '+ssc' in spec else 'OFF'))
 
+        if self.spec.version >= Version('2.5.0'):
+            args.append('-DADIOS2_USE_DataSpaces={0}'.format(
+                'ON' if '+dataspaces' in spec else 'OFF'))
+
+        # transitive dependencies in SST via DILL, FFS, EVPath, etc.
+        if spec.satisfies('~sst'):
```

Comment:
None of these will get searched for if is disabled so `ffi` and `fabric` don't need to be disabled at all and rather than turning off the others unconditionally, they can be dependent on sst:
``` python
if '+sst' in spec:
    args.extend([
        # Broken dependency package
        '-DCMAKE_DISABLE_FIND_PACKAGE_BISON=TRUE',

        # Depends on ^
        '-DCMAKE_DISABLE_FIND_PACKAGE_FLEX=TRUE',

        # Not yet packaged
        '-DCMAKE_DISABLE_FIND_PACKAGE_CrayDRC=TRUE',
        '-DCMAKE_DISABLE_FIND_PACKAGE_NVSTREAM=TRUE'
    ])
```
```

Reply:
You are right, that's so much simpler. Patch pushed.